### PR TITLE
Rewrite and modularization of image types, how to verify, how to unco…

### PR DESCRIPTION
**Pull request by bktan8**
…mpress..

On 2017-11-30:
--------------
1) Created the download-image.rst file in the Guides/Maintaince.  This is a hidden file and not visible under the Maintenance tree.  This talks about types of Clear Linux images, how to verify checksum, and how to uncompress for Linux, macOS, and Windows.  This is the base for other docs such as the bare-metal, live, bootable USBs (Linux, macOS, Windows), VMware docs...
2) Reworked the bootable-usb-linux.rst to re-use content from download-image.rst.
3) Reworked the bootable-usb-mac.rst to re-use content from download-image.rst.
4) Reworked the bootable-usb-windows.rst to re-use content from download-image.rst.
5) Reworked the live-image.rst to re-use content from bootable-usb-linux.rst and download-image.rst.
6) Reworked the bare-metal doc to re-use content from bootable-usb-linux.rst and download-image.rst.

On 2017-12-01:
--------------
1) Updated the types of images list and created a table.
2) Remove references to the old "releases" repo.  Users should only use the "image" folder - per engineering.

On 2017-12-05:
--------------
1) Recaptured the 7zip screenshot and made it look better.
2) Added "download-image.rst" to the Guides/Maintenance/maintenance.rst" tree to make it visible.
3) Moved the "Download the latest Clear Linux image" out of the "download-image.rst" and put it into
   the "bootable-usb-linux.rst" and tagged it so it could be used by "bootable-usb-mac.rst" and
   "bootable-usb-windows.rst".

On 2017-12-07:
--------------
1) Changed the "Types of images" list Guides/Maintenance/download-image.rst to a table format.

On 2017-12-13:
--------------
1) Changed all instances of <> to [] to indicate substitution.
2) Added |CAUTION-BACKUP-USB| in substitution.rst.  This is used in create-usb-linux.rst, create-usb-mac.rst, create-usb-windows.rst.
3) Deleted download-image.rst and created 3 new docs, one for each OS: download-verify-uncompress-linux.rst, download-verify-uncompress-mac.rst, download-verify-uncompress-windows.rst.
4) General fixes based on feedback.

On 2017-12-18:
--------------
1) Moved the "Types of Images" section to its own doc. called "types-of-cl-images.rst".
2) Added "types-of-cl-images" to Guides/Maintenance/maintenance.rst" TOC tree.
3) Updated bare-metal-install.rst, live-image.rst, bootable-usb-linux.rst, bootable-usb-mac.rst, bootable-usb-windows.rst, download-verify-uncompress-linux.rst, download-verify-uncompress-mac.rst, download-verify-uncompress-windows.rst to include "types-of-cl-images".
4) Excluded instruction to uncompress GZ file from bootable-usb-linux.rst because we're only dealing with XZ file.
5) Expanded on the bootable-usb-windows.rst instruction on how to use Rufus with a few more screenshots.

Signed-off-by: Bun K Tan <bun.k.tan@intel.com>

### CODE REVIEW

**Comment by bktan8** [comment]
@rcaballeromx - Even though the "bktan1-usb-rewrite" branch started out in the master repo, my last edits to it was done on my own fork.  So when I pushed the changes to my fork, the original PR and comments were not included in this new PR.  Hope this is not a problem.  

I updated this branch to include your suggestions.

**Comment by bktan8** [comment]
@rcaballeromx - I will keep the original PR in the main repo open until you tell me otherwise.

**Comment by rcaballeromx** [comment]
@bktan8 Let's talk about pull requests tomorrow when we meet. I think this could be beneficial to more than just yourself. It will also be a lot quicker than if I type a lengthy explanation here.

**Comment by bktan8** [comment]
@rcaballeromx - ready for your review again.

File: source/clear-linux/get-started/bootable-usb/bootable-usb-mac.rst [hunk 1]
```diff
@@ -11,75 +11,70 @@ Alternative instructions for other operating systems are available:
...
 
-      $ sudo dd if=/path/to/clear-<version>-live.img of=/dev/rdisk2 bs=4m
+      $ sudo dd if=./clear-[version number]-[image type] of=/dev/rdisk2 bs=4m
 
-   * Adding an ‘r’ in front of the disk identifier should help speed up the
-     imaging process.
-   * Use `<CTL>-T` to check imaging progress.
+   .. note::
+   
+      * Adding an ‘r’ in front of the disk identifier should help speed up the
```
Comment (rcaballeromx) [inline source/clear-linux/get-started/bootable-usb/bootable-usb-mac.rst]: These two must not be in a list. They are already inside a note.

File: source/clear-linux/get-started/bootable-usb/bootable-usb-windows.rst [hunk 2]
```diff
@@ -11,46 +11,57 @@ Alternative instructions for other operating systems are available:
...
+      :alt: Rufus utility - Show and select |CL| image file
+
+      Figure 2: Rufus utility - Show and select |CL| image file
+
+#. Verify that :guilabel:`Create a bootable disk using DD Image` checkbox is checked. 
+   See Figure 3.
 
-#. Select the |CL| image file and ensure that “Create a bootable disk using DD
-   Image” is checked. See Figure 1.
+   .. figure:: figures/Rufus-figure-3.png
```
Comment (rcaballeromx) [inline source/clear-linux/get-started/bootable-usb/bootable-usb-windows.rst]: please respect the filename conventions for figures.

File: source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+=====================================
+
+Go to the Clear Linux `image`_ repository and download the desired type of image.
+
+.. _verify-image-checksum-on-windows:
+
+Verify the integrity of the Clear Linux image
+=============================================
+
+Before you use a downloaded |CL| image, it is recommended that you verify its
```
Comment (rcaballeromx) [inline source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst]: Do not use "it" nor over-complicated language:
... |CL| image, verify its integrity ...

File: source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+Go to the Clear Linux `image`_ repository and download the desired type of image.
+
+.. _verify-image-checksum-on-windows:
+
+Verify the integrity of the Clear Linux image
+=============================================
+
+Before you use a downloaded |CL| image, it is recommended that you verify its 
+integrity to eliminate the chance, albeit small, that it might be corrupted due
```
Comment (rcaballeromx) [inline source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst]: Keep language simple, concise, and clear:
integrity. Thus, you eliminate the small chance of a corrupted image due to download issues.

File: source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+Go to the Clear Linux `image`_ repository and download the desired type of image.
+
+.. _verify-image-checksum-on-windows:
+
+Verify the integrity of the Clear Linux image
+=============================================
+
+Before you use a downloaded |CL| image, it is recommended that you verify its 
+integrity to eliminate the chance, albeit small, that it might be corrupted due 
+to any download issues.  To support this, each released |CL| image is accompanied
```
Comment (rcaballeromx) [inline source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst]: Do not use "this" without an appropriate noun and avoid passive voice:
To support verification, we accompany each released |CL| image with an official ...

File: source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+#.  Manually compare the output with the original checksum value shown in 
+    the downloaded checksum file and make sure they match.
+
+.. _uncompress-image-on-windows:
+
+Uncompress the Clear Linux image
+================================
+
+All released |CL| images are compressed by default using either GNU zip
```
Comment (rcaballeromx) [inline source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst]: Switch to active voice:
We compress all released |CL| images by default with either GNU ...

File: source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    the downloaded checksum file and make sure they match.
+
+.. _uncompress-image-on-windows:
+
+Uncompress the Clear Linux image
+================================
+
+All released |CL| images are compressed by default using either GNU zip 
+(`.gz`) or XZ (`.xz`).  The compression typed used is dependent on which 
+platform/enviroment the image is targeted for.  Uncompress your image using
```
Comment (rcaballeromx) [inline source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst]: Keep language simple:
The compression type we use depends on the target platform or environment of the image.

File: source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+.. _uncompress-image-on-windows:
+
+Uncompress the Clear Linux image
+================================
+
+All released |CL| images are compressed by default using either GNU zip 
+(`.gz`) or XZ (`.xz`).  The compression typed used is dependent on which 
+platform/enviroment the image is targeted for.  Uncompress your image using 
+these steps:
```
Comment (rcaballeromx) [inline source/clear-linux/guides/maintenance/download-verify-uncompress-windows.rst]: Consistency:
To uncompress the image, follow these steps:

File: source/clear-linux/guides/maintenance/types-of-cl-images.rst [hunk 9]
```diff
@@ -0,0 +1,53 @@
+.. _types-of-cl-images:
+
+Types of Clear Linux images
+***************************
```
Comment (rcaballeromx) [inline source/clear-linux/guides/maintenance/types-of-cl-images.rst]: The top level heading must be underlined with ###

**Comment by bktan8** [comment]
@rcaballeromx - ready for another review.  Thx!

**Comment by rcaballeromx** [comment]
I've created a new pull request with all my fixes and pointing to the correct branch for merging: #77

**Pull request closed without merging**